content/en/blog/_posts/2023-01-12-protect-mission-critical-pods-priorityclass/index.md

```diff
@@ -0,0 +1,300 @@
+
+## Resource management in Kubernetes 
+
+The control plane consists of multiple components, out of which the scheduler (usually the built-in [kube-scheduler](/docs/concepts/scheduling-eviction/kube-scheduler/)) is one of the components which is responsible for assigning a node to a pod.
+- Whenever a pod is created, it goes to a pending state, and then the scheduler kicks in and determines which node should be the best-suited node for the placement of the new pod.
+  - In the background, the scheduler runs as an infinite loop looking for pods without a `nodeName` set
+    (and that are [ready for scheduling](/docs/concepts/scheduling-eviction/pod-scheduling-readiness/)).
+  - For each Pod that needs scheduling, the scheduler tries to make its best decision about which node should run that Pod.
+  - To name a few, `nodeSelector` , `taints and tolerations` , `nodeAffinity` , the rank of nodes based on available resources (for example, CPU and memory), and several other criteria are used to determine the pod's placement.  
+- Just in case the scheduler is not able to find any node, the pod remains to be in the pending state, which is not an ideal scenario.
+
+The below diagram explains how the request flows in (starting from point number 1 to 4):
+
+{{< figure src=kube-scheduler.svg alt="A diagram showing the scheduling of three Pods that a client has directly created." title="Scheduling in Kubernetes">}}
+
+## Typical use cases
+
+Below are some real-life scenarios where control over the scheduling and eviction of pods may be required.
+
+1. Let's say the pod you are planning to deploy is really critical, and you have some resource constraints. How will you make the resources available in the cluster? by first identifying which pods are not required, then manually deleting them, etc., which is a manual and tedious process, or you may plan to add a new node to the cluster, which too might take some time and money. 
+
+   An example of such a case is a DaemonSet of an infrastructure component like Grafana Loki. In that case, the Loki pods must run on each node before any other pods run on that node.
+
+2. Another use case can be: Suppose, you have a single cluster, and that cluster holds the pods for the below environments:
+   - preprod [Priority 1] 
+   - staging [Priority 2] 
+   - development [Priority 3]
+
+   Unfortunately, the cluster goes through high resource consumption, and now there is a resource competition on the nodes with respect to the CPU and the memory. At least in the short term, you don’t have further nodes to expand or scale that Kubernetes cluster (cluster level autoscaling _may_ add more nodes, but this takes time). Perhaps some pods will remain in a Pending state, or perhaps service will degrade as Pods compete for resources. If the kubelet does evict a Pod from a node, that eviction happens at random because the kubelet doesn’t have any special information about which Pods to evict and which to keep.
```

Comment:
```suggestion
In the event of high resource consumption on the cluster, there is competition for CPU and memory resources on the nodes. While cluster-level autoscaling _may_ add more nodes, it takes time.  In the interim, if there are no further nodes to scale the cluster, some Pods could remain in a Pending state, or service could be degraded as they compete for resources. If the kubelet does evict a Pod from the node, that eviction would be random because the kubelet doesn’t have any special information about which Pods to evict and which to keep.
```